assets/js/components/StatesBarChart.vue

```diff
@@ -365,17 +365,52 @@ export default Chart.extend({
         }, {})
     },
 
+    /**
+     * returns projects/programmes for donors/beneficiaries
+     * if a donor or beneficiary is selected (filter is applied)
+     * it will show the projects/programmes for that state
+     */
     legend_items() {
       // just the totals merged into the types
       const out = {}
+
       for (const t in this.types) {
-        out[t] = Object.assign({
-          value: this.totals[t],
-        }, this.types[t])
+        if(this.filters[this.state_type]) {
+          out[t] = Object.assign({
+            value: this.filter_values[t],
+          }, this.types[t])
+        } else {
+          out[t] = Object.assign({
+            value: this.totals[t],
+          }, this.types[t])
+        }
       }
+
       // return an array though
       return d3.values(out)
     },
+
+    /**
```

Comment:
this comment is verbose, unneeded, and wrong. :)
it says it returns 3 things, neither of which is true.